client/dashboard/domains/domain-contact-details/contact-form.tsx

```diff
@@ -0,0 +1,223 @@
+import { useMutation, useQuery } from '@tanstack/react-query';
+import {
+	ExternalLink,
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+	__experimentalText as Text,
+	Button,
+	Card,
+	CardBody,
+} from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
+import { DataForm, Field, isItemValid } from '@wordpress/dataviews';
+import { createInterpolateElement } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+import { store as noticesStore } from '@wordpress/notices';
+import { useEffect, useRef, useState, useMemo } from 'react';
+import { countryListQuery, statesListQuery } from '../../app/queries/domain-supported-contries';
+import { domainWhoisMutation } from '../../app/queries/domain-whois';
+import Notice from '../../components/notice';
+import { fetchDomainWhoisValidate, type DomainContactDetails } from '../../data/domain-whois';
+import { getContactFormFields } from './contact-form-fields';
+
+import './contact-form.scss';
+interface ContactFormProps {
+	domainName: string;
+	initialData?: DomainContactDetails;
+	onSubmit?: ( data: DomainContactDetails ) => void;
+	onCancel?: () => void;
+	errors?: Partial< Record< keyof DomainContactDetails, string > >;
+}
+
+export default function ContactForm( {
+	domainName,
+	initialData,
+	onSubmit,
+	onCancel,
+}: ContactFormProps ) {
+	const { createSuccessNotice, createErrorNotice } = useDispatch( noticesStore );
+	const { data: countryList } = useQuery( countryListQuery() );
+	const [ selectedCountryCode, setSelectedCountryCode ] = useState(
+		initialData?.countryCode ?? ''
+	);
+	const { data: statesList } = useQuery( statesListQuery( selectedCountryCode ) );
+
+	const formDataRef = useRef< any >( null );
+	const updateMutation = useMutation( domainWhoisMutation( domainName ) );
+
+	const validateMutation = useMutation( {
+		mutationFn: ( formData: any ) => {
```

Comment:
We should reuse the existing `domainWhoisValidateQuery()` here. And I believe it should then be a mutation, not query?

Reply:
> using a POST request to fetch its data? 😕😢

My suspicion is that the payload is so large that using GET might be inconvenient.

> Maybe the function should be called validateDomainWhois() rather than fetchDomainWhoisValidate().

Agree 👍